__tests__/commands/install/resolutions.js

```diff
@@ -92,3 +95,14 @@ test.concurrent('install with nested resolutions using flat mode', (): Promise<v
     expect(await getPackageVersion(config, 'ansi-regex')).toEqual('1.1.1');
   });
 });
+
+test.concurrent('install with resolutions does not write new lockfile if existing one satisfied', (): Promise<void> => {
+  return runInstall(
+    {},
+    {source: 'resolutions', cwd: 'install-with-resolutions-dont-write-new-lockfile-if-satisfied'},
+    async (config): Promise<void> => {
+      const lockfile = await fs.readFile(path.join(config.cwd, 'yarn.lock'));
+      expect(lockfile.indexOf('foobar')).toBeGreaterThanOrEqual(0);
+    },
+  );
+});
```

Comment:
Can you rather test that we correctly bailout (ie `install with resolution settings should correctly bailout during the integrity check`)?

Reply:
done